datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -3541,6 +3595,12 @@ mod tests {
         ];
 
         assert_optimized!(expected, plan.clone(), true);
+
+        let expected = &[
```

Comment:
This behavior change is a regression from IOx's perspective as the plan is now resorting data that is already sorted. I will work on creating a reproducer / fix next week

Reply:
This plan is generated when the `bounded_order_preserving_variants` configuration flag is `false`. When this flag is `true`, we get the sort-free result. The prior behavior was basically ignoring/overriding the flag.

For some detailed context: The flag basically lets the user choose whether they want `SortPreservingMerge`s, or `Repartition`/`Coalesce`+`Sort` cascades. We ran some benchmarks and there is no clearly dominating strategy, each alternative comes out ahead in certain cases. In non-streaming cases, the first alternative typically came out ahead, so we let the default flag value to be `false`.

Since we are a stream-first platform, we set the flag to `true` at Synnada. Maybe IOx also wants to set this flag to `true`?

Reply:
Ah, thank you for the hint. I will give it a try

Reply:
Update here is that this suggested worked well -- thank you. I find the naming of `bounded_order_preserving_variants` to be confusing so I have proposed a new name here https://github.com/apache/arrow-datafusion/pull/7723